### PR TITLE
Refactor Plausible.Billing.Plans module

### PR DESCRIPTION
**Pull request by vinibrsl**
This pull request introduces a series of improvements to Plausible.Billing.Plans, including:

- Tag the JSON file with the plan version
- Rename the JSON field `limit` to `monthly_pageview_limit`
- Move site_limit function to Billing.Plans
- Refactor subscription_interval, allowance and site_limit functions
- Remove unused AnalyzePlans task

### CODE REVIEW

File: lib/plausible/billing/plans.ex [hunk 1]
```diff
@@ -96,47 +105,87 @@ defmodule Plausible.Billing.Plans do
...
+
+      _any ->
+        "N/A"
     end
   end
 
-  def allowance(%Plausible.Billing.Subscription{paddle_plan_id: "free_10k"}), do: 10_000
-
+  @spec allowance(Plausible.Billing.Subscription.t()) :: non_neg_integer() | nil
   def allowance(subscription) do
```
Comment (RobertJoonas) [inline lib/plausible/billing/plans.ex:169]: How about renaming `allowance` to `monthly_pageview_limit` everywhere in code as well? Might make it clearer that it's the same thing

**Review by RobertJoonas — APPROVED** [review]
LGTM! Two small suggestions inline

File: lib/plausible/billing/plans.ex [hunk 2]
```diff
@@ -96,47 +105,87 @@ defmodule Plausible.Billing.Plans do
...
+
+    case get_subscription_plan(user.subscription) do
+      %Plausible.Billing.EnterprisePlan{} -> :unlimited
+      %Plausible.Billing.Plan{site_limit: site_limit} -> site_limit
+      :free_10k -> @site_limit_for_free_10k
+      nil -> @site_limit_for_trials
+    end
+  end
+
+  defp get_subscription_plan(%Plausible.Auth.User{} = user) do
```
Comment (ukutaht) [inline lib/plausible/billing/plans.ex:138]: I don't see this variant being used, and tests pass when deleted. Is it needed?

**Review by ukutaht — APPROVED** [review]

File: lib/plausible/billing/plans.ex (hunk 2 above)
Comment (vinibrsl) [reply lib/plausible/billing/plans.ex]: Good call!

File: lib/plausible/billing/plans.ex (hunk 1 above)
Comment (vinibrsl) [reply lib/plausible/billing/plans.ex:164]: Good call!

**Pull request merged**